src/wallet/wallet.cpp

```diff
@@ -4303,7 +4303,7 @@ int CWallet::ScanForWalletTransactions(
 
         // no need to read and scan block, if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        while (pindex && nTimeFirstKey && pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
+        while (chainActive.Next(pindex) != NULL && nTimeFirstKey && pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
             pindex = chainActive.Next(pindex);
         }
```

Comment:
```suggestion
        if (nTimeFirstKey) {
            while (pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
                CBlockIndex* pindex_next = chainActive.Next(pindex);
                if (pindex_next == nullptr) break;
                pindex = pindex_next;
            }
        }
```
(same effect, may be slightly more efficient)

Reply:
i think i'm going to keep my version, because i don't see any efficiency benefits (the compiler probably compiles everything down to the same code) and it's more clear that it's doing a traverse as long as the three conditions hold